inst/include/Rcpp/date_datetime/Date.h

```diff
@@ -102,6 +102,16 @@ namespace Rcpp {
             return m_d;
         }
 
+        inline std::string format(const char *fmt = "%Y-%m-%d") const {
+            char txt[32];
+            struct tm temp = m_tm;
+            temp.tm_year -= baseYear();    // adjust for fact that system has year rel. to 1900
+            ::strftime(txt, 31, fmt, &temp);
+            return std::string(txt);
```

Comment:
Since `txt` isn't zero-initialized (it's a plain C array), it seems like it could be possible that no null byte exists at the end of `txt`. Or does `::strftime()` write a null byte at the end? What if the output of `::strftime()` doesn't fit in `txt`?